### DISCUSSION TITLE
DisplaCy for Dataframe

### DISCUSSION THREAD

**Discussion by BenWaka** [explosion/spaCy#10800] [Help: Coding & Implementations]
how can I visualize using displaCy in a dataframe? I have a data called "taks_output" and want to visualize a sample of the columm "msg_lower"? What I did:

```
import pandas as pd
import numpy as np
import spacy 
import nltk
from spacy import displacy

df = task_output['msg_lower']
df = df.sample(n=1787)
displacy.render(df,  style='dep', jupyter=True, options={'distance': 90})
```
And i get this: "Invalid object passed to displaCy: Can only visualize Doc or Span objects, or dicts if set to manual=True."

What can i do?

**Comment by polm** [comment]
Like the error says, you can only pass Docs or Spans to displacy. I'm not sure where your data is but you need to get it out of the dataframe to give it to spaCy. spaCy has no special integration with dataframes.

**Reply by BenWaka** [reply]
> Like the error says, you can only pass Docs or Spans to displacy. I'm not sure where your data is but you need to get it out of the dataframe to give it to spaCy. spaCy has no special integration with dataframes.

But how can I do this?

**Reply by polm** [reply]
I don't use dataframes much but I assume you can do something like this:

```
texts = df. ... something to get a list of strings ...

for doc in nlp.pipe(texts):
    displacy.render(doc, style="dep")
```
